utils/tra_utils.py

```diff
@@ -0,0 +1,213 @@
+from pathlib import Path
+import re
+from typing import TextIO, Sequence, Mapping, Set, Tuple, List, Optional, Union, Type, TypeVar
+
+
+Self = TypeVar('Self')
+
+class WeiduFile:
+    name: str
+    path: Path
+    io: TextIO
+
+    subroots: Set[str] = {
+        'audio',
+        'areas',
+        'creatures',
+        'dialogues',
+        'epilogues',
+        'items',
+        'lib',
+        'scripts',
+        'spells',
+        'translations',
+    }
```

Comment:
Maybe those could be auto detected from folder structure?

Reply:
Potentially, by detecting a .tp2 file and an .exe file placed "above" and named the same as the tp2's directory... which seemed slightly more complex on the first try, but may be a better option at the end of the day, as that's the real community convention.

Problem being --- we're using the "translations directory" convention already, which isn't accepted in all community mods. To do it properly, we'd need to have a look at the .tp2 and check the AUTO_TRA... which is a bit much work for such a simple python script (but maybe it's a good idea to have it, eventually...)